src/lib/mathlib/math/filter/NotchFilter.hpp

```diff
@@ -99,8 +123,26 @@ class NotchFilter
 		b[2] = _b2;
 	}
 
+	/**
+	 * Bypasses the filter update to directly set different filter coefficients.
+	 * Note: the filtered frequency and quality factor saved on the filter lose their
+	 * physical meaning if you use this method to change the filter's coefficients.
+	 * Used for creating clones of a specific filter.
+	 */
+	void setCoefficients(float a[2], float b[3])
+	{
+		_a1 = a[0];
+		_a2 = a[1];
+		_b0 = b[0];
+		_b1 = b[1];
+		_b2 = b[2];
+	}
+
+
 	T reset(const T &sample);
 
+	void update(float sample_freq, float notch_freq, float bandwidth);
```

Comment:
@francelico I just noticed the actual update() code is missing.

Reply:
I think the `update` isn't required as we can update the coefficients using `setParameters`

Reply:
Hi, yes you are correct @bresch this method is not used anywhere else, it can be deleted. @dagar how should I proceed for the fix, should I make a new PR (and mention this one in the description)?

Reply:
I'll delete it quickly if there's no intention of using it. Thanks for the followup. https://github.com/PX4/Firmware/pull/16061